### ISSUE TITLE
fix auto build also on development_calc + fix examples

### ISSUE THREAD

**Issue by despiegk** [crystaluniverse/crystallib#134]
## examples

* examples can be under each module
* when using a main() make sure you have only 1 file per directory (the examples can have multiple subdirs)

## issues with auto test

* seem to be some issues https://github.com/freeflowuniverse/crystallib/actions
* fix for development_calc and others
* merge development_calc in branches which are not development or main (if possible)
* check autotests are working
* do we do the tests on the auto build?
* if issues on tests lets fix

**Comment by brandonpille** [comment]
The tests are working on development. If development_calc is an old branch it should be rebased on development. I would not merge development_calc into branches other than development or main. That's how strange merge issues occur. Please rebase on development and merge into development once it is done. 

This is the proper way of using git. Branch off of development (or some other branch) and merge into the same branch (development or the branch you branched off of) once done. It's not a good idea to merge branch into branches that have diverged from development.

**Comment by brandonpille** [comment]
The tests may have been failing last week because of a bug that was introduced in the V compiler: https://github.com/vlang/v/issues/17689#event-8784514894. That has been fixed so tests should be green again. There has also been a change in the http package in V. http.new_request no longer returns a optional result. But that has been fixed on development too.

**Comment by brandonpille** [comment]
I merged development into the branch development_calc which makes it possible now to merge the branch into development. But I see a lot of changes in this branch. If I open a PR I see a lot of change that will be introduced into development. What exactly is in this branch? Is there an issue related to it? If not I would suggest to not work like that in the future. Branches should not diverge too much from development for a couple of reasons:
- It's easy to merge the branch into development
- It's easier to review the PR once you want to merge into development
- There is less chance to break something in the exisisting code
- We can release faster
Some good guidelines:
- Create issues that are smaller
- Branch off of development and merge into development once done (not into another branch)
- Add tests and make sure they are green before landing the PR
